### PR TITLE
ekf2: migrate fuse_airspeed to SymForce

### PR DESCRIPTION
**Pull request by bresch**
## Describe problem solved by this pull request
Copy/paste and manually modifying auto-generated code is prone to human errors; it takes about a day of painful work to replace the existing auto-generated code if anything in the derivation is changed.

## Describe your solution
Use SymForce to generate compatible C++ code and only use function calls within PX4

## Test data / coverage
SITL test with `make px4_sitl gazebo_plane__windy`

Before:
DeepinScreenshot_select-area_20220930142648

This PR:
DeepinScreenshot_select-area_20220930142632

### CODE REVIEW

**Comment by dagar** [comment]
I added a little cmake helper to re-run the Symforce code generation. By default it shouldn't do anything.

File: src/modules/ekf2/EKF/airspeed_fusion.cpp [hunk 1]
```diff
@@ -43,45 +43,29 @@
...
-			       math::constrain(airspeed_sample.eas2tas, 0.9f, 10.0f));
-
-	// Intermediate variables
-	const float IV0 = ve - vwe;
-	const float IV1 = vn - vwn;
-	const float IV2 = (IV0)*(IV0) + (IV1)*(IV1) + (vd)*(vd);
-
-	const float predicted_airspeed = sqrtf(IV2);
+	const float R = sq(math::constrain(_params.eas_noise, 0.5f, 5.0f) *
+			   math::constrain(airspeed_sample.eas2tas, 0.9f, 10.0f));
```
Comment (dagar) [inline src/modules/ekf2/EKF/airspeed_fusion.cpp:59]: 10 seems kind of high for eas2tas, I think that would be > 30 km.

File: src/modules/ekf2/EKF/airspeed_fusion.cpp [hunk 2]
```diff
@@ -150,31 +109,24 @@ void Ekf::fuseAirspeed(estimator_aid_source_1d_s &airspeed)
...
-	SparseVector24f<4,5,6,22,23> Hfusion;
-	Hfusion.at<4>() = HK4;
-	Hfusion.at<5>() = HK5;
-	Hfusion.at<6>() = HK6;
-	Hfusion.at<22>() = HK7;
-	Hfusion.at<23>() = HK8;
+	Vector24f H; // Observation jacobian
+	Vector24f K; // Kalman gain vector
+
+	matrix::Vector<float, 24> state_vector(reinterpret_cast<const float*>(&_state));
```
Comment (dagar) [inline src/modules/ekf2/EKF/airspeed_fusion.cpp:117]: This is still a copy into state_vector.

I would just do the dumb/simple/safe/verbose version for now until we're ready to make a more sweeping change. 

state_vector(0) = _state.quat_nominal(0), etc

Comment (bresch) [reply src/modules/ekf2/EKF/airspeed_fusion.cpp:117]: It's actually already implemented: https://github.com/PX4/PX4-Autopilot/blob/20b46ad33f0a64ad6beabe5a8ab2a27fd3bdee9b/src/modules/ekf2/EKF/ekf_helper.cpp#L624-L636

**Comment by bresch** [comment]
Added a unit test to compare the previous auto-generated code with the new functions to make sure the results are numerically identical.

**Review by dagar — APPROVED** [review]

**Pull request merged**